lib/datadog/tracing/contrib/action_pack/action_dispatch/instrumentation.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+require_relative '../../../metadata/ext'
+
+module Datadog
+  module Tracing
+    module Contrib
+      module ActionPack
+        module ActionDispatch
+          # Instrumentation for ActionDispatch components
+          module Instrumentation
+            module_function
+
+            def set_http_route_tag(http_route)
+              return if http_route.nil?
+
+              active_span = Tracing.active_span
+              return unless active_span
+
+              active_span.set_tag(
+                Tracing::Metadata::Ext::HTTP::TAG_ROUTE,
+                http_route.gsub(/\(.:format\)$/, '')
+              )
+            end
+
+            # Instrumentation for ActionDispatch::Journey components
+            module Journey
+              # Instrumentation for ActionDispatch::Journey::Router
+              # for Rails versions older than 7.1
+              module Router
+                def find_routes(req)
+                  result = super(req)
+
+                  return response unless Tracing.enabled?
+
+                  begin
+                    # Journey::Router#find_routes retuns an array for each matching route.
+                    # This array is [match_data, path_parameters, route].
+                    # We need the route object, since it has a path with route specification.
+                    current_route = result.last&.last&.path&.spec
+                    return result unless current_route
+
+                    # When Rails is serving requests to Rails Engine routes, this function is called
+                    # twice: first time for the route on which the engine is mounted, and second
+                    # time for the internal engine route.
+                    last_route = Tracing.active_span&.get_tag(Tracing::Metadata::Ext::HTTP::TAG_ROUTE)
+                    Instrumentation.set_http_route_tag(last_route.to_s + current_route.to_s)
```

Comment:
Is it still intended/ok to call `set_http_route_tag` when `active_span` is nil?

Reply:
good point, I will add an early return instead of a safe operator